pkg/backends/influxdb/flush.go

```diff
@@ -0,0 +1,256 @@
+			}
+			sb.WriteRune(ch)
+		}
+	}
+}
+
+// escape will escape NL (\n), CR (\r), TAB (\t), space, comma, and backslash,
+// writing the results to the supplied strings.Builder.
+func escapeNameToBuilder(sb *strings.Builder, s string) {
+	for _, ch := range s {
+		if ch == '\n' {
+			sb.WriteString("\\n")
+		} else if ch == '\r' {
+			sb.WriteString("\\r")
+		} else if ch == '\t' {
+			sb.WriteString("\\t")
+		} else {
+			if ch == ' ' || ch == ',' || ch == '\\' {
+				sb.WriteByte('\\')
+			}
+			sb.WriteRune(ch)
+		}
+	}
+}
+
+// escapeStringToBuilder will escape double quotes and backslash writing the
+// results to the supplied strings.Builder, surrounded by double-quotes
+func escapeStringToBuilder(sb *strings.Builder, s string) {
+	sb.WriteByte('"')
+	for _, ch := range s {
+		if ch == '\\' || ch == '"' {
+			sb.WriteByte('\\')
+		}
+		sb.WriteRune(ch)
+	}
+	sb.WriteByte('"')
+}
+
+// writeName writes "measurement,tag1=value1,tag2=value2,... " to the output buffer.
+// Common to all metric types and events.
+func writeName(w io.Writer, name string, tags gostatsd.Tags) {
+	_, _ = w.Write([]byte(formatNameTags(name, tags)))
+}
+
+func (f *flush) addCounter(name string, tags gostatsd.Tags, count int64, rate float64) {
+	writeName(f.writer, name, tags)
+	_, _ = f.writer.Write([]byte(fmt.Sprintf("count=%d,rate=%g %d\n", count, rate, f.timestampSeconds)))
+	f.metricCount++
+	f.maybeFlush()
+}
+
+func (f *flush) addGauge(name string, tags gostatsd.Tags, value float64) {
+	writeName(f.writer, name, tags)
+	_, _ = f.writer.Write([]byte(fmt.Sprintf("value=%g %d\n", value, f.timestampSeconds)))
+	f.metricCount++
+	f.maybeFlush()
+}
+
+func (f *flush) addSet(name string, tags gostatsd.Tags, value uint64) {
+	writeName(f.writer, name, tags)
+	_, _ = f.writer.Write([]byte(fmt.Sprintf("count=%d %d\n", value, f.timestampSeconds)))
+	f.metricCount++
+	f.maybeFlush()
+}
+
+func (f *flush) addBaseTimer(name string, timer gostatsd.Timer) {
+	var sb strings.Builder
+	if !f.disabledSubtypes.Lower {
+		sb.WriteString(fmt.Sprintf("lower=%g,", timer.Min))
+	}
+	if !f.disabledSubtypes.Upper {
+		sb.WriteString(fmt.Sprintf("upper=%g,", timer.Max))
+	}
+	if !f.disabledSubtypes.Count {
+		sb.WriteString(fmt.Sprintf("count=%d,", timer.Count))
+	}
+	if !f.disabledSubtypes.CountPerSecond {
+		sb.WriteString(fmt.Sprintf("rate=%g,", timer.PerSecond))
+	}
+	if !f.disabledSubtypes.Mean {
+		sb.WriteString(fmt.Sprintf("mean=%g,", timer.Mean))
+	}
+	if !f.disabledSubtypes.Median {
+		sb.WriteString(fmt.Sprintf("median=%g,", timer.Median))
+	}
+	if !f.disabledSubtypes.StdDev {
+		sb.WriteString(fmt.Sprintf("stddev=%g,", timer.StdDev))
+	}
+	if !f.disabledSubtypes.Sum {
+		sb.WriteString(fmt.Sprintf("sum=%g,", timer.Sum))
+	}
+	if !f.disabledSubtypes.SumSquares {
+		sb.WriteString(fmt.Sprintf("sum_squares=%g,", timer.SumSquares))
+	}
+	for _, pct := range timer.Percentiles {
+		sb.WriteString(fmt.Sprintf("%s=%g,", pct.Str, pct.Float))
+	}
+	if sb.Len() == 0 { // wat
```

Comment:
Nice comment m8

Reply:
Who am I to question if some sysadmin has decided to disable all timers!